packages/core/server/admin-api/services/url-pattern.ts

```diff
@@ -38,47 +33,43 @@ export default () => ({
    */
   findOne: async (id: number) => {
     const patternEntity = await strapi.entityService.findOne('plugin::webtools.url-pattern', id);
-
     return patternEntity;
   },
-
   /**
    * FindByUid.
    *
    * @param {string} uid the uid.
    * @param {string} langcode the langcode.
    */
-  findByUid: async (uid: string, langcode?: string): Promise<string> => {
+  findByUid: async (uid: string, langcode?: string): Promise<string[]> => {
     let patterns = await getPluginService('urlPatternService').findMany({
       filters: {
         contenttype: uid,
       },
     });
 
     if (langcode) {
-      patterns = patterns
-        .filter((pattern) => (pattern.languages as string).includes(langcode));
+      patterns = patterns.filter((pattern) => (pattern.languages as string).includes(langcode));
     }
 
-    if (!patterns[0]) {
-      return strapi.config.get('plugin.webtools.default_pattern');
+    if (! patterns.length) {
+      return [strapi.config.get('plugin.webtools.default_pattern')];
     }
 
-    return patterns[0].pattern;
-  },
+    const patterns2 = patterns.map((pattern) => pattern.pattern);
```

Comment:
I think we should just be able to return that map in this function, instead of creating the const `patterns2`.